insights/combiners/limits_conf.py

```diff
@@ -0,0 +1,60 @@
+"""
+Limits configuration
+====================
+
+The limits files are normally available to rules as a list of LimitsConf
+objects.  This combiner turns those into one set of data, and provides a
+``find_all()`` method to search the rules from all the files.
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.limits_conf import LimitsConf
+
+
+@combiner(requires=[LimitsConf])
+class AllLimitsConf(object):
+    """
+    Combiner for accessing all the limits configuration files.
+
+    Attributes:
+        domains(set): the set of domains found in all data files.
+        limits(list): a list of the original LimitsConf parser results.
+        rules(list): the entire list of rules.
+    """
+    def __init__(self, local, shared):
+        rules = []
+        domains = set([])
+        self.limits = shared[LimitsConf]
+        for limits in shared[LimitsConf]:
+            # The rule dictionary contains the file_path as a key for later.
+            rules.extend(limits.rules)
```

Comment:
Good question.  There's no mention of how this is handled in the limits.conf man page, apart from:

> In general, individual limits have priority over group limits, so if you impose no limits for admin group, but one of the members in this group have a limits line, the user will have its limits set according to this line.

I believe that the files are read in name order, which is why some limits files are numbered - e.g. '90-nproc.conf'.  However, there's no requirement for files to not contain duplicate limits or for particular limits to be in any particular file - e.g. 'nproc' limits can be defined anywhere, not just in the '90-nproc.conf' file.

We can enforce that order in that loop by making line 28 something like:

`for limits in sorted(shared[LimitsConf], key=lambda x: getattr(x, 'file_path')):`

But @jhjaggars or @kylape may know if the files are presented in name order, which would mean that we default to the 'correct' way of working.

De-duplicating this list is going to be complicated, because we can't compare on the entire row, we have to compare on only the domain, type and item of each row.  A temporary list of that subset would make that possible.

However, the `find_all()` method is going to give people all the limits that match the query.  In most cases that should be a list with a single row, but the matching function is designed to handle the matching rules defined in `man limits.conf` - e.g matching on user, group, user ID range, and more.  Rules which use this output will need to handle multiple rows, which means that the will need to handle duplicates.

Aside - we could provide a method like find_all called, say, `find_most_specific` which attempts to implement the logic of selecting the most specific limit that affects a particular domain and type.  But I'm not sure yet if this is going to be used by any rule so I'm loath to add that complexity.